harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -787,19 +1206,60 @@ def _load(self, load_path: pathlib.Path) -> None:
                 logging.warning(
                     f"Callback '{name}' implements load_state_dict(), but no callback state "
                     "was found for that name when restoring from checkpoint. This "
-                    "callback will be initialized from scratch"
+                    "callback will be initialized from scratch."
                 )
 
-        # Load workload sequencer state.
-        wlsq_path = load_path.joinpath("workload_sequencer.pkl")
-        if self.wlsq is not None and wlsq_path.exists():
-            with wlsq_path.open("rb") as f:
-                self.wlsq.load_state(pickle.load(f))
+        save_path = load_path.joinpath("trial_state.pkl")
+        if save_path.exists():
+            with save_path.open("rb") as f:
+                self._load_state(pickle.load(f))
+        else:
+            # Support legacy save states
+            wlsq_path = load_path.joinpath("workload_sequencer.pkl")
+            if wlsq_path.exists():
+                with wlsq_path.open("rb") as f:
+                    self._load_wlsq_state(pickle.load(f))
```

Comment:
Do you have a test to make sure this codepath works?

This is a codepath that strikes me as 95% useless if it doesn't work the day you ship it, because users will see a break across this upgrade transition and this codepath won't be useful thereafter.

Reply:
(I don't think this needs to block landing the PR because I don't care all that much about this transition codepath, but I do think if you land this now you should try to land the test before the release ships)